### DISCUSSION TITLE
NEW UPDATES 🎉  Micro wakeword 2 - Timer visuals - Few tweaks etc.

### DISCUSSION THREAD

**Discussion by BigBobbas** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#44] [Announcements]
Hi All ,
I have been doing some work over the last day or so to enhance the config, I wasn't planning on releasing it just yet, but as there is a breaking change... I thought I would release this to the dev -branch with the required change in place. Please treat as experimental although it is working really well for me.

I have made the following changes. I hope you find them useful and it makes the box even better. changes are as follows.
### **EDIT:** 🎉 
**a. I have just  added the function to display STT & TTS Response on screen - this can be toggled on and off in the voice settings page.**  
### **END_OF_EDIT:** 🎉

1. Removed battery icon from status bar and replaced with a 'Timers' state icon. red = no timers green= timer in progress. Tapping the icon will show a page with the timers remaining time, if it is active (only supports the first timer at the moment),  A page will display with the last ten seconds countdown when it reaches that point. 

2. added wifi connection details to a 'wifi page' when you tap the wifi icon.

3. added battery level to the info page ( will add a new page in next update with a battery level warning when the battery is low )

4. Fixed breaking change caused by Micro Wake Word 2 - mww2 will now allow you to use multiple on device wakewords at once. please see the info [here](<https://esphome.io/components/micro_wake_word.html>)

If you have any issues or suggestions , ideas, contribution is most welcome.
### **Links to configs:**
Sensor version [here](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/dev/s3b_dev.yaml>)
Non Sensor [here](<https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/dev/s3b_dev_no_sensors.yaml>)

Thanks for supporting the project everyone 🙏 

Best Wishes

Rob

**Comment by Someguitarist** [comment]
Currently getting this with the sensor version when attempting to install over ESPhome:

INFO Generating C++ source...
INFO Updating https://github.com/espressif/esp-adf.git@v2.5
INFO Updating submodules (components/esp-adf-libs, components/esp-sr) for https://github.com/espressif/esp-adf.git@v2.5
Traceback (most recent call last):
  File "/usr/local/bin/esphome", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/esphome/esphome/__main__.py", line 1079, in main
    return run_esphome(sys.argv)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/esphome/esphome/__main__.py", line 1066, in run_esphome
    rc = POST_CONFIG_ACTIONS[args.command](args, config)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/esphome/esphome/__main__.py", line 484, in command_run
    exit_code = write_cpp(config)
                ^^^^^^^^^^^^^^^^^
  File "/esphome/esphome/__main__.py", line 194, in write_cpp
    return write_cpp_file()
           ^^^^^^^^^^^^^^^^
  File "/esphome/esphome/__main__.py", line 212, in write_cpp_file
    writer.write_cpp(code_s)
  File "/esphome/esphome/writer.py", line 344, in write_cpp
    copy_src_tree()
  File "/esphome/esphome/writer.py", line 297, in copy_src_tree
    copy_files()
  File "/esphome/esphome/components/esp32/__init__.py", line 684, in copy_files
    repo_dir, _ = git.clone_or_update(
                  ^^^^^^^^^^^^^^^^^^^^
  File "/esphome/esphome/git.py", line 111, in clone_or_update
    run_git_command(
  File "/esphome/esphome/git.py", line 31, in run_git_command
    raise cv.Invalid(lines[-1][len("fatal: ") :])
voluptuous.error.Invalid: Unable to find current revision in submodule path 'components/esp-adf-libs'

**Reply by BigBobbas** [reply]
this sometimes happens, try clicking 'retry' sometimes you have to do it a few times. If that doesn't help try restarting ESPHome add-on

**Reply by Someguitarist** [reply]
Just to follow up, retrying a few times did fix it!  Thanks!

**Comment by MetalOnes** [comment]
Awesome updates, appreciate all your work! It took a little time to merge the updates with my customizations, but it wasn't too bad.
I have encountered some freezes, mostly when poking around the new Settings pages, but it's been solid for the most part.

Also - you should add a BuyMeACoffee link or something to your profile ;)

**Comment by pepe59** [comment]
There is an error in this block when editing it below and it won't compile

```
#### voice settings page #####
  - platform: touchscreen
    page_id: ww_set_page
    id: ww_chang_hass
    internal: true
    x_min: 10
    x_max: 315
    y_min: 180
    y_max: 160
    on_click:
      min_length: 10ms
      max_length: 500ms
      then:
        - select.operation:
            id: wake_word_engine_location
            operation: Next
            cycle: true   
        - component.update: s3_box_lcd
        # - script.execute: saver_enabled
```
y_max is less than y_min
Values ​​are swapped

**Reply by BigBobbas** [reply]
HI thanks for pointing that at, which config was this from ?

Many Thanks 

Rob

**Comment by BigBobbas** [comment]
Sorry , I found it... that was my error when made changes this morning.
I have now fixed 🤞 

Thanks for helping and pointing this out 👍

**Comment by pepe59** [comment]
I don't know if this is expected behavior, but the name is missing from the block and therefore the switch is not listed in HA

```
  - platform: template
    name: Screensaver
    id: s_saver
    optimistic: true    
    restore_mode: RESTORE_DEFAULT_OFF
     
  - platform: template
    id: s_saver_mode
    optimistic: true    
    restore_mode: RESTORE_DEFAULT_OFF
```
s3b_dev.yaml

**Reply by BigBobbas** [reply]
ah yes... I started to tidy up the not required entities showing in HA ... still in the process, but those 2 should have stayed. Thanks for pointing this out. I do need to find some time to go through everything and tidy it all up etc. And also need to streamline things as it's getting difficult to manage.
Thanks you again for helping it is greatly appreciated. 👍 
I will add them back in now.

**Comment by pepe59** [comment]
Everything works great as expected.
The only thing I found is that after compiling it immediately connects to the HA api and it is not enough to load the screen (touch) and the device has to be physically rebooted sometimes several times.
I'm studying the whole code to solve this and so far I'm wondering if the problem is in the block

```
  on_boot:
    priority: 600
```

**Comment by BigBobbas** [comment]
this is an odd one, as I don't get this issue at all, however others have.

The onboot should only effect these... which shouldn't cause any issues... 
```yaml
      - lambda: id(led).turn_on().set_brightness(1.0).perform();
      - display.page.show: loading_page
      - component.update: s3_box_lcd
      - lambda: id(time_remaining_0).publish_state ("0:00:00"); 
      - script.execute: saver_enabled
      ```
      however it may be as you say .. something in the priority of the startup that can cause it. If you watch the logs over USB from https://web.esphome.io and click the 'reset device' link on the logs page... it will show the full boot process. from bootloader to esphome start and device setup. Maybe something in there would give a clue.
      
The i2c bus provides connectivity to the AHT temp/hum sensor, the touchscreen and the DAC for audio output. maybe the DAC may cause this, i'm not sure if perhaps that could be delayed on startup.

**Comment by pepe59** [comment]
Thanks, I'll see if I can find anything in the logs

**Reply by BigBobbas** [reply]
both of my devices are working so it's difficult for me to troubleshoot. I bought them 6 months apart and was hoping there would be some difference , but both have same pcb revision and the only differences are some slight variation to the housing . 

So yes if you could view the  logs... that would be a great help
